src/Http/Organizer/LegacyOrganizerRequestBodyParser.php

```diff
@@ -21,20 +21,46 @@ public function parse(ServerRequestInterface $request): ServerRequestInterface
         }
         $data = clone $data;
 
-        $data->url = $data->website;
+        $mainLanguage = $data->mainLanguage ?? null;
 
-        $data->name = [
-            $data->mainLanguage => $data->name,
-        ];
+        if (isset($data->website) && !isset($data->url)) {
+            $data->url = $data->website;
+            unset($data->website);
+        }
 
-        if (isset($data->address)) {
-            $data->address = [
-                $data->mainLanguage => $data->address,
+        if ($mainLanguage && isset($data->name) && is_string($data->name)) {
+            $data->name = (object) [
+                $data->mainLanguage => $data->name,
             ];
         }
 
-        if (isset($data->contact)) {
+        if ($mainLanguage && isset($data->address->streetAddress)) {
+            $data->address->{$mainLanguage} = $data->address->{$mainLanguage} ?? (object) [];
```

Comment:
This can maybe only be done once and removed out of the condition?

Reply:
No because then it sometimes happens that you just get an empty object. I did that at first and a test complained about it :)

Reply:
I could extract this logic that's in in each `if` to a method thought but I wasn't sure how important this is since it's not really business logic but some data transformation. Plus it's for a legacy schema that shouldn't change anymore. So as long as tests cover it it I thought it's not very important?